atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/transaction/impl/SnapshotTransaction.java

```diff
@@ -785,10 +786,15 @@ private boolean isValidationNecessary(TableReference tableRef) {
                 tableRef,
                 rangeRequest.getStartInclusive(),
                 endRowExclusive).entrySet();
-        return ImmutableList.copyOf(mergeInLocalWrites(
+
+        // This filters out empty values.
+        ImmutableList<Entry<Cell, byte[]>> result = ImmutableList.copyOf(mergeInLocalWrites(
                 postFilteredCells.iterator(),
                 localWritesInRange.iterator(),
                 rangeRequest.isReverse()));
+        getPostFilteredCellsMeter().mark(postFilteredCells.size() + localWritesInRange.size() - result.size());
```

Comment:
i think this collection contains the set of results to return to the client, but we are actually interested in the number of cells that were filtered out (I guess `postFilteredCells` is kind of an ambiguous name here).

Reply:
note that `postFilteredCells.size()` is going to be `O(n)` as its a filtered collection -- we might want to eagerly copy.